cli/util/upload_bin.go

```diff
@@ -340,41 +345,36 @@ func uploadBlocks(ctx *cli.Context, p *pool.Pool, rpc *rpcclient.Client, signer
 
 // uploadIndexFiles uploads missing index files to the container.
 func uploadIndexFiles(ctx *cli.Context, p *pool.Pool, containerID cid.ID, account *wallet.Account, signer user.Signer, oldestMissingBlockIndex uint, blockAttributeKey string, homomorphicHashingDisabled bool, maxParallelSearches, maxRetries int, debug bool) error {
-	attributeKey := ctx.String("index-attribute")
-	indexFileSize := ctx.Uint("index-file-size")
-	fmt.Fprintln(ctx.App.Writer, "Uploading index files...")
-
-	prm := client.PrmObjectSearch{}
-	filters := object.NewSearchFilters()
-	filters.AddFilter(attributeKey, fmt.Sprintf("%d", 0), object.MatchNumGE)
-	filters.AddFilter("IndexSize", fmt.Sprintf("%d", indexFileSize), object.MatchStringEqual)
-	prm.SetFilters(filters)
-	var objectIDs []oid.ID
-	errSearch := retry(func() error {
-		var errSearchIndex error
-		objectIDs, errSearchIndex = neofs.ObjectSearch(ctx.Context, p, account.PrivateKey(), containerID.String(), prm)
-		return errSearchIndex
-	}, maxRetries)
-	if errSearch != nil {
-		return fmt.Errorf("index files search failed: %w", errSearch)
-	}
-
-	existingIndexCount := uint(len(objectIDs))
-	expectedIndexCount := (oldestMissingBlockIndex - 1) / indexFileSize
-	if existingIndexCount >= expectedIndexCount {
-		fmt.Fprintf(ctx.App.Writer, "Index files are up to date. Existing: %d, expected: %d\n", existingIndexCount, expectedIndexCount)
-		return nil
-	}
-	fmt.Fprintf(ctx.App.Writer, "Current index files count: %d, expected: %d\n", existingIndexCount, expectedIndexCount)
 	var (
+		attributeKey  = ctx.String("index-attribute")
+		indexFileSize = ctx.Uint("index-file-size")
```

Comment:
Some variables here... Not sure I follow the logic. In general, the scope should be minimized.